### PR TITLE
chore: clean the gutter 

### PR DESCRIPTION
**Pull request by Reverendyz**

### CODE REVIEW

File: backend/src/asset/asset.service.ts [hunk 1]
```diff
@@ -5,29 +5,79 @@ import { AssetDto } from './dto';
...
 export class AssetService {
-    constructor(
-        private prisma: PrismaService,
-    ){}
-    async addToWallet(user: User, asset: AssetDto){
-        const value = Number(asset.value)
-        const wallet = await this.prisma.wallet.findUnique({
+  constructor(
+    private prisma: PrismaService,
+    ) { }
```
Comment (Gustavopereira42) [inline backend/src/asset/asset.service.ts]: The indentation looks weird here... maybe something like this?
```suggestion
  constructor(
    private prisma: PrismaService,
  ) {}
```
or it could be inline
```suggestion
  constructor(private prisma: PrismaService) {}
```

File: backend/src/asset/asset.service.ts [hunk 2]
```diff
@@ -5,29 +5,79 @@ import { AssetDto } from './dto';
...
+              value: totalValue,
+              averagePrice: averagePrice,
+            },
+          });
+  
+          return {
+            success: true,
+            message: 'Asset quantity updated successfully.'
+          };
+        } else {
```
Comment (Gustavopereira42) [inline backend/src/asset/asset.service.ts]: We don't need the else here since the if above already returns something

File: frontend/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        "parser": "@typescript-eslint/parser"
+    },
+    "plugins": [
+        "@typescript-eslint",
+        "vue"
+    ],
+    "rules": {
+        "indent": [
+            "error",
+            4
```
Comment (Gustavopereira42) [inline frontend/.eslintrc.json]: Maybe keep it "2" here because we're using 2 spaces both in the backend and the HTML?

File: frontend/.eslintrc.json
Comment (Gustavopereira42) [inline frontend/.eslintrc.json:1]: That looks good, we could also include eslint in the backend 👀 
We can make this in another PR, this one is already big enough

File: frontend/src/router/index.ts [hunk 4]
```diff
@@ -1,30 +1,72 @@
...
-export default router
+router.beforeEach((to, from, next) => {
+    const token = sessionStorage.getItem('access_token'); 
+    if (token) {
+        const endpoint = '/auth/validate';
+        fetch(process.env.VUE_APP_API_URL + endpoint, {
+            method: 'POST',
+            headers: {
+                'Authorization': `Bearer ${token}`
+            }
```
Comment (Gustavopereira42) [inline frontend/src/router/index.ts]: We can create a file (or more) to centralize the authenticated API calls. We can define the base API URL and the headers in just one place. A Library like [axios](https://axios-http.com/) could help.

Later we can also use the store `actions` to handle with API calls

File: frontend/src/router/index.ts [hunk 5]
```diff
@@ -1,30 +1,72 @@
...
+        }).then(response => {
+            if (response.ok) {
+                next();
+            } else {
+                localStorage.removeItem('token');
+                return {
+                    name: 'NotFound'
+                };
```
Comment (Gustavopereira42) [inline frontend/src/router/index.ts]: Ideally, we must always use the `next()` function inside the router beforeEach. If the user does not have a token or it is invalid we should redirect them to the login page using `next({ name: 'home' })`

**Review by Gustavopereira42 — CHANGES_REQUESTED** [review]

File: frontend/src/router/index.ts (hunk 4 above)
Comment (Reverendyz) [reply frontend/src/router/index.ts]: Let's talk about that. It seems interesting, in fact. About axios, I was trying not to include too many libraries that I found "unnecessary". 
As you are the js expert, feel free to give us hints

File: frontend/src/router/index.ts (hunk 5 above)
Comment (Reverendyz) [reply frontend/src/router/index.ts]: That was the thing I was struggling today. Because I wanted to check the token or the state `isLogged` and not return to the home screen.

But I was trying to understand how does VUE manage this.

Especially if the user IS VALID and how to proceed? `next(to)`?

Comment (Gustavopereira42) [reply frontend/src/router/index.ts]: The `beforeEach` will be executed **before** the user goes to any route, so what you do here must always send the user to some route. If you call `next()` without params it will continue to the next route. If you wanna send the user to another route depending on some condition, you can call the `next` using the route params, like `next({ name: 'foo', params: { bar: true } })`. 

In this case, if the user is valid you can just use `next()`.

File: frontend/src/router/index.ts (hunk 4 above)
Comment (Gustavopereira42) [reply frontend/src/router/index.ts]: I think it's worth it. It's a small library and includes some features that will help a lot in terms of reuse code and there are some cool features like an "Interceptor" to handle any API calls that fail for some reason. We also can easily use a refresh token to increase the security of user authentication.

Comment (Reverendyz) [reply frontend/src/router/index.ts]: I will create a task to do this while this PR is merged. Then we can address this feature easily. Thanks for the input.

**Review by Gustavopereira42 — APPROVED** [review]
LGTM
